src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -925,21 +928,29 @@ def aks_enable_addons(cmd, client, resource_group_name, name, addons,
         result = LongRunningOperation(cmd.cli_ctx)(
             client.begin_create_or_update(resource_group_name, name, instance))
 
-        # For monitoring addon, Metrics role assignement doesnt require in case of MSI auth
-        if enable_monitoring and not enable_msi_auth_for_monitoring:
+        if enable_monitoring:
+            from msrestazure.tools import resource_id
+            cluster_resource_id = resource_id(
+                subscription=subscription_id,
+                resource_group=resource_group_name,
+                namespace='Microsoft.ContainerService', type='managedClusters',
+                name=name
+            )
             cloud_name = cmd.cli_ctx.cloud.name
+            # For monitoring addon, Metrics role assignement doesnt require in case of MSI auth
+            if not enable_msi_auth_for_monitoring and cloud_name.lower() == 'azurecloud':            
             # mdm metrics supported only in Azure Public cloud so add the role assignment only in this cloud
-            if cloud_name.lower() == 'azurecloud':
-                from msrestazure.tools import resource_id
-                cluster_resource_id = resource_id(
-                    subscription=subscription_id,
-                    resource_group=resource_group_name,
-                    namespace='Microsoft.ContainerService', type='managedClusters',
-                    name=name
-                )
                 add_monitoring_role_assignment(
                     result, cluster_resource_id, cmd)
 
+            if enable_msi_auth_for_monitoring and cloud_name.lower() != 'azurecloud':
+                try: 
+                    get_user_assigned_identity_by_resource_id(cluster_resource_id)
```

Comment:
Should the import statement for this function added just before the use? Similar to what is there for resource_id function  on line 932